smt/surrogate_models/krg_based.py

```diff
@@ -130,7 +180,9 @@ def _new_train(self):
             self._specific_train()
         else:
             if self.options["eval_noise"]:
-                self.optimal_theta = self.optimal_theta[:-1]
+                if not self.options["is_noise_het"]:
+                    self.noise = self.optimal_theta[self.D.shape[1] :]
```

Comment:
Why the case heteroscedastic is not treated like the homoscedastic one ?
Can't we uniformize these two ?
Just by adding the length ?

Reply:
We can do it but that forces the initialisation of the noise variances in a list, p.e. : noise0 = nt * [1e-6] ... and check the option using len(noise).